elasticdl/python/worker/main.py

```diff
@@ -24,12 +25,27 @@ def main():
             channel = build_channel(addr)
 
             # Wait the channel is ready by a Future object.
-            grpc.channel_ready_future(channel).result()
-            logger.info(
-                "grpc channel %s to connect pod %s is ready"
-                % (addr, addr.split(".")[0])
-            )
-            ps_channels.append(channel)
+            connect_succeed = False
+            for i in range(3):
+                try:
```

Comment:
Also would be good explaining why this is needed. Why not simply increasing the timeout instead of retrying with timeout=60?

Reply:
The worker may start before PS and the future will be waiting even if the PS starts. So, the worker can retry to connect with the PS after it starts.

Reply:
`grpc.channel_ready_future` is waiting until the GRPC server channel is ready. It is ok for PS to launch after worker pods.
Why need to add timeout and retry? Is this for the case that when PS pods may fail to run,  worker pods can exit?

Reply:
In many experiments, I find some workers hang and always wait GRPC channel is ready. After I delete the worker, the new worker succeeds to connect with PS.